src/TimeService/timeservice-api.c

```diff
@@ -0,0 +1,85 @@
+#include "timeservice-api.h"
+
+static uint64_t starting_time;
+
+void nanoseconds_to_cuc_timestamp(uint64_t nanoseconds, CUCTimestamp *timestamp)
+{
+   long double seconds = (long double)nanoseconds / NS_PER_SECOND;
+   uint64_t seconds_shifted = seconds * 256 * 256 * 256;
+
+   ADJUST_ENDIANESS(seconds_shifted);
+   unsigned char *time = (unsigned char *)&seconds_shifted;
+
+   CUCTimestamp *timestamp_struct = (CUCTimestamp *)timestamp;
+
+   for (size_t i = 0; i < sizeof(timestamp_struct->tField); i++)
+   {
+      timestamp_struct->tField[i] = time[sizeof(timestamp_struct->tField) - 1 - i];
+   }
+}
+
+void TimeService_Startup(void)
+{
+    INIT_TIMERS();
+    starting_time = get_clock();
+}
+
+void TimeService_CfsTimestampCmp(const CfsTimestamp *t1, const CfsTimestamp *t2, int *result)
+{
+    if (t1->seconds > t2->seconds)
+    {
+        *result = 1;
+    }
+    else if (t1->seconds < t2->seconds)
+    {
+        *result = -1;
+    }
+    else if (t1->seconds == t2->seconds)
+    {
+        if (t1->subseconds > t2->subseconds)
+        {
+            *result = 1;
+        }
+        else if (t1->subseconds < t2->subseconds)
+        {
+            *result = -1;
+        }
+        else
+        {
+            *result = 0;
+        }
+    }
+}
+
+
+void TimeService_ClockStatus(ClockStatusEnum *status)
+{
+   *status = STOPPED;
+
+   uint64_t start_time = get_clock();
+   delay(10000);
```

Comment:
is waiting 10 cycles good idea to check clock status?

Reply:
I will try to use clock_nanosleep and rtems version of it but the whole function is not well described in D5 and I am not sure what that function should actually do and return

Reply:
I still dont understand why we check clock status this way. Using delay to check if clock is running seems a big anti-pattern to me. Is not there any flag or can't we introduce one to check for clock status?